pkg/resolver/client/ens/ens.go

```diff
@@ -121,10 +139,44 @@ func (c *Client) Close() error {
 	return nil
 }
 
-func wrapResolve(backend bind.ContractBackend, name string) (string, error) {
+func wrapDial(endpoint string, contractAddr string) (*ethclient.Client, *goens.Registry, error) {
```

Comment:
note for this method and the next - would be nice to wrap the errors before returning them, this way we know where the error came from. examples below

Reply:
Very good point!

Reply:
There is no need to include the function name in the Errorf messages. This is up to the caller to decide. I think that all `wrapdial: ` can be removed. I see that this was suggested by @acud and adjusted accordingly. This is just a suggestion.